src/PaginatedCollectionFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace G4\Collection;
+
+use G4\DataRepository\DataRepositoryResponse;
+use G4\DataRepository\Exception\MissingResponseAllDataException;
+use G4\Factory\ReconstituteInterface;
+
+class PaginatedCollectionFactory
+{
+    /**
+     * @throws MissingResponseAllDataException
+     */
+    public static function create(
+        DataRepositoryResponse $response,
+        ReconstituteInterface  $factory,
+        Pagination             $pagination
+    ): PaginatedCollection {
+        $collection = new PaginatedCollection($response->getAll(), $factory);
+        $collection
+            ->setItemsCountPerPage($pagination->getPerPage())
+            ->setCurrentPageNumber($pagination->getPage())
+            ->setPageCount((int)$pagination->calculatePageCount($response->getTotal()))
+            ->setTotalItemsCount((int)$response->getTotal())
+            ->setCurrentItemsCount($response->count());
+        return $collection;
+    }
+
+    public static function createEmpty(ReconstituteInterface $factory, Pagination $pagination): PaginatedCollection
+    {
+        $collection = new PaginatedCollection([], $factory);
+        $collection
+            ->setCurrentItemsCount($collection->getCurrentItemsCount())
```

Comment:
Instantiate `$collection = new PaginatedCollection([], $factory)` and then `getCurrentItemsCount()` and `getTotalItemsCount()` which both return `null` (because of the bug inside `PaginatedCollection`.
Set them to 0 because we know that the collection is empty.

Reply:
Ispravio sam na projektu... nisam preneo izmenu na paket. Ima default vrednosti kolekcija kao smo pricali. ide azuriranje

Reply:
OK je sada posle izmene na https://github.com/g4code/collection/pull/4/commits/77392450b8eb3d377ea9db858edeb8933fe027c1
Malo je redudantno da kreiras kolekciju i onda da toj kolekciji setujes `setCurrentItemsCount( $collection->getCurrentItemsCount())`

Ili stavi 0 umesto `$collection->getCurrentItemsCount()` i `$collection->getTotalItemsCount()` ili ukloni ta dva setera za currentItemsCount i totalItemsCount